src/axom/core/tests/core_array.hpp

```diff
@@ -2389,102 +2389,52 @@ TEST(core_array, reserve_nontrivial_reloc_um)
 }
 #endif
 
-// Regression test for a memory leak in axom::Array's copy/move/initialization
-template <typename T1, typename T2>
-auto make_arr_data(int SZ1, int SZ2)
-  -> std::pair<axom::Array<axom::Array<T1>>, axom::Array<axom::Array<T2>>>
+class AllocatingDefaultInit
 {
-  using Arr1 = axom::Array<axom::Array<T1>>;
-  using Arr2 = axom::Array<axom::Array<T2>>;
+public:
+  AllocatingDefaultInit() { m_value.resize(32); }
 
-  T1 val1 {};
-  T2 val2 {};
-
-  Arr1 arr1(SZ1, 2 * SZ1);
-  for(int i = 0; i < SZ1; ++i)
-  {
-    arr1[i].resize(i + 10);
-    arr1[i].shrink();
-    arr1[i].push_back(val1);
-  }
-
-  Arr2 arr2(SZ2, 2 * SZ2);
-  for(int i = 0; i < SZ2; ++i)
-  {
-    arr2[i].resize(i + 10);
-    arr2[i].shrink();
-    arr2[i].push_back(val2);
-  }
-  arr2.shrink();
-
-  return {std::move(arr1), std::move(arr2)};
-}
+private:
+  axom::Array<int> m_value;
+};
 
-TEST(core_array, regression_array_move)
```

Comment:
@kennyweiss -- I replaced the regression test you added in #1286 with a unit test that I think should capture all the leaky cases you were seeing. Let me know if you'd rather keep the regression test as well.

Reply:
Thanks @publixsubfan -- it looks like it covers similar/more ground, so I'm ok w/ removing the old regression case.

Did you check the bugfix against lsan and/or valgrind or a similar tool?